src/snyk/common/analytics/itly.ts

```diff
@@ -136,144 +145,156 @@ export class Iteratively implements IAnalytics {
+    });
   }
 
   public logIssueHoverIsDisplayed(properties: IssueHoverIsDisplayedProperties): void {
-    if (!this.canReportEvents() || !this.user.authenticatedId) {
-      return;
-    }
-
-    itly.issueHoverIsDisplayed(this.user.authenticatedId, properties);
+    this.enqueueEvent(() => {
+      itly.issueHoverIsDisplayed(this.getAuthenticatedUserId(), properties);
+    });
   }
 
   public logScanModeIsSelected(properties: Omit<ScanModeIsSelectedProperties, 'eventSource' | 'ide'>): void {
-    if (!this.canReportEvents() || !this.user.authenticatedId) {
-      return;
-    }
+    this.enqueueEvent(() => {
+      itly.scanModeIsSelected(this.getAuthenticatedUserId(), {
+        ...properties,
+        ide: this.ide,
+        eventSource: 'IDE',
+      });
+    });
+  }
 
-    itly.scanModeIsSelected(this.user.authenticatedId, {
-      ...properties,
-      ide: this.ide,
-      eventSource: 'IDE',
+  public logBackgroundAnalysisNotificationIsDisplayed(
+    properties: Omit<BackgroundAnalysisNotificationIsDisplayedProperties, 'eventSource' | 'ide'>,
+  ): void {
+    this.enqueueEvent(() => {
+      itly.backgroundAnalysisNotificationIsDisplayed(this.getAuthenticatedUserId(), {
+        ...properties,
+        ide: this.ide,
+        eventSource: 'IDE',
+      });
     });
   }
 
-  public logFalsePositiveIsDisplayed(): void {
-    if (!this.canReportEvents() || !this.user.authenticatedId) {
-      return;
-    }
+  public logBackgroundAnalysisNotificationButtonIsClicked(
+    properties: Omit<BackgroundAnalysisNotificationButtonIsClickedProperties, 'eventSource' | 'ide'>,
+  ): void {
+    this.enqueueEvent(() => {
+      itly.backgroundAnalysisNotificationButtonIsClicked(this.getAuthenticatedUserId(), {
+        ...properties,
+        ide: this.ide,
+        eventSource: 'IDE',
+      });
+    });
+  }
 
-    itly.falsePositiveIsDisplayed(this.user.authenticatedId, {
-      ide: this.ide,
-      eventSource: 'IDE',
+  public logFalsePositiveIsDisplayed(): void {
+    this.enqueueEvent(() => {
+      itly.falsePositiveIsDisplayed(this.getAuthenticatedUserId(), {
+        ide: this.ide,
+        eventSource: 'IDE',
+      });
     });
   }
 
   public logFalsePositiveIsSubmitted(
     properties: Omit<FalsePositiveIsSubmittedProperties, 'eventSource' | 'ide'>,
   ): void {
-    if (!this.canReportEvents() || !this.user.authenticatedId) {
+    this.enqueueEvent(() => {
+      itly.falsePositiveIsSubmitted(this.getAuthenticatedUserId(), {
+        ...properties,
+        ide: this.ide,
+        eventSource: 'IDE',
+      });
+    });
+  }
+
+  private enqueueEvent(eventFunction: () => void, mustBeAuthenticated = true): void {
+    if (mustBeAuthenticated && !this.user.authenticatedId) {
       return;
     }
 
-    itly.falsePositiveIsSubmitted(this.user.authenticatedId, {
-      ...properties,
-      ide: this.ide,
-      eventSource: 'IDE',
-    });
+    if (!this.canReportEvents()) {
```

Comment:
I'd pull this up to be the first guard, as it would overrule authentication check.